src/client.ts

```diff
@@ -45,6 +45,16 @@ export function createClient(
     },
   });
 
+  admin.interceptors.response.use(
+    function (response) {
+      if (response.data.errors) {
+        throw new Error(`Shopify admin API responded with error: ${JSON.stringify(response.data.errors)}`)
+      }
```

Comment:
I am hesitant to merge this as is since there may be errors we don't want to throw for. Thinking we could throw on `"message":"access denied"` and log a warning for other cases.

Reply:
I think in most cases where something comes back in `data.errors`, you would not want to continue with sourcing as it likely indicates something is going to be missing in the response, which will break either the source plugin code (like it did in this case) or the Gatsby queries further down the stack.

It really depends on if you want to continue sourcing or not once an error has been returned. If you did want to continue anyway, outputting a warning with the contents of the error would at least give the developer an idea of why it failed further down the stack.

I'm happy with either solution 🙂